kerboscript_tests/functest31.ks

```diff
@@ -0,0 +1,38 @@
+//
+// I am not too sure if this is a feature we want to advertise yet,
+// but it does seem to be doable:
+//
+print "Testing varying args logic".
+
+// A function that accepts varying args.
+// the first arg is a number: how many more args to expect.
+// the rest of the args are just printed for testing.
+function print_var_args {
+  parameter how_many.
+  
+  // read parameters, filling a local string.
+  local my_string is "".
+  local i is 1.
+  until i > how_many {
+    parameter next_arg.
+    set my_string to my_string + " " + next_arg.
+    set i to i + 1.
+  }
```

Comment:
OOPS - in writing up the above description, a problem case just occurred to me!

Imagine this code:
```
function foo
{
    parameter p1.
    if p1 = 0 { return. }
    parameter p2.
    // other code here.
}
```

I think that will break, big time.  Because when the ``return`` statement kicks in, it circumvents the OpcodeArgBottom that happens after the ``parameter p2.``, and thus, leaving junk atop the stack, causes stack misalignment.

I'll have to think on the problem a bit longer.